OCP-OCP migration (#411)
kubev2v/forklift-documentation: documentation/modules/migrating-virtual-machines-cli.adoc

Migration using {osp} source providers only supports VMs that use only Cinder volumes.
====

[NOTE]
====

> You can create `Provider` manifests for future use if the virtual machines that they include currently lack storage.

Reviewer: I still don't understand what this means.. that's related to the empty mappings we talked about before, right?
in that case, I'd suggest to add to `network: <3>` below a note that says something like: network mapping should be specified also if the vms to be migrated don't have networks, the mapping can be empty in this case.

btw, I'm looking at `network: <3>` below and seems we need to change the current description of this item ("You can add multiple network mappings") - the plan can point to a single network (or storage) mapping

Author: @ahadas  about changing "You can add multiple network mappings" Are you saying we allow one network map per plan? If so, do I need to state this for storage below (no callout now).

Reviewer: @RichardHoch correct. the network map can map multiple source networks to destination networks

> You can create `Provider` manifests for future use if the virtual machines that they include currently have no storages.

Reviewer 2: ```suggestion
Create manifests for future use if the virtual machines they include currently lack storage.
```

this is a tenuous comment - please ignore if you feel so minded

    destination:
      name: <destination_cluster>
      namespace: {namespace}
  map: <3>
    network: <4>
      name: <network_map> <5>
      namespace: {namespace}
    storage:
      name: <storage_map> <6>
      namespace: {namespace}
  targetNamespace: {namespace}
  vms: <7>
    - id: <source_vm> <8>
    - name: <source_vm>
      namespace: {namespace} <9>
      hooks: <10>
        - hook:
            namespace: {namespace}
            name: <hook> <11>
          step: <step> <12>
EOF
----
<1> Specify the name of the `Plan` CR.
<2> Specify whether the migration is warm or cold. If you specify a warm migration without specifying a value for the `cutover` parameter in the `Migration` manifest, only the precopy stage will run.
<3> Specify only one network map and one storage map per plan.

> <4> Specify a network mapping even if the VMs to be migrated are not assigned to a network. The mapping can be empty in this case.

Reviewer: cool, and we need a similar note for the storage mapping

Author: Done.

  vms: <6>
    - id: <source_vm> <7>
    - name: <source_vm>
      namespace: {namespace} <8>
      hooks: <9>
        - hook:
            namespace: {namespace}
            name: <hook> <10>
          step: <step> <11>
EOF
----
<1> Specify the name of the `Plan` CR.
<2> Specify whether the migration is warm or cold. If you specify a warm migration without specifying a value for the `cutover` parameter in the `Migration` manifest, only the precopy stage will run.
<3> You can add multiple network mappings.
<4> Specify the name of the `NetworkMap` CR.
<5> Specify the name of the `StorageMap` CR.

> <6> For all source providers except for {virt}, you can use either the `id` _or_ the `name` parameter to specify the source VMs. +

Reviewer 3: `namespace` needs to be specified as well

Reviewer: yeah, that's (8)
@RichardHoch I think we need to consider replacing this part with an example per-provider as it starts to become complicated to have many exceptions per-item

Author: @ahadas @anarnold97 I think having an example per provider is a good idea. Can it wait until after GA?

Reviewer 2: @RichardHoch - I will have to defer to Arik, but I think the example is a good idea.
Do you think you could get it done in your post-release clean-up to minimize the delay?
Thanks

Author: @ahadas @anarnold97 We are talking about this section https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.4/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#migrating-virtual-machines-from-cli

@ahadas  Did you mean we should have examples per step (that is, all the examples of Secret manifest in step 1, all the the examples of Provider manifests in step 2, etc.) or instructions per provider type (that is RHV manifests from start to finish in one section, vSphere from start to finish in another)? 

In theory, i should be able to create all the manifests from what's written here and have QE check them. 

In reply to another PR, Arik suggested that we reorganize the document by provider, as in...

1. vSphere migrations
  a. UI
  b. CLI
  
2. RHV migrations
 a. UI
 b. CLI 
 
That might be a good long-term idea.

Reviewer: > @ahadas Did you mean we should have examples per step (that is, all the examples of Secret manifest in step 1, all the the examples of Provider manifests in step 2, etc.) or instructions per provider type (that is RHV manifests from start to finish in one section, vSphere from start to finish in another)?

instructions per provider type

> 
> In theory, i should be able to create all the manifests from what's written here and have QE check them.
> 
> In reply to another PR, Arik suggested that we reorganize the document by provider, as in...
> 
> 1. vSphere migrations
>    a. UI
>    b. CLI
> 2. RHV migrations
>    a. UI
>    b. CLI

yeah, in those (b) sections

  vms: <6>
    - id: <source_vm> <7>
    - name: <source_vm>
      namespace: {namespace} <8>
      hooks: <9>
        - hook:
            namespace: {namespace}
            name: <hook> <10>
          step: <step> <11>
EOF
----
<1> Specify the name of the `Plan` CR.
<2> Specify whether the migration is warm or cold. If you specify a warm migration without specifying a value for the `cutover` parameter in the `Migration` manifest, only the precopy stage will run.
<3> You can add multiple network mappings.
<4> Specify the name of the `NetworkMap` CR.
<5> Specify the name of the `StorageMap` CR.

> <6> For all source providers except for {virt}, you can use either the `id` _or_ the `name` parameter to specify the source VMs. +

Reviewer 2: Nice... the only problem is that the others are not as well written!
